docs/DeveloperGuide.md

```diff
@@ -695,22 +726,103 @@ testers are expected to do more *exploratory* testing.
+      ```
+      add n/Benjamin low 
+      p/98765431 
+      e/benLow@gmail.com 
+      s/NUS Global Merit Scholarship 
+      as/pending 
+      m/Medicine 
+      m/Mathematics
+      ```
+      Expected: An applicant named `Benjamin Low` with the above characteristics
+                will be added to the applicant list and displayed on TrackAScholar GUI.
+      
+   1. Test case:
+      ```
+      add n/Alex Yeoh 
+      p/88712345 
+      e/alexYeoh@gmail.com 
+      s/NUS Merit Scholarship 
+      as/pending 
+      m/Computer Science
+      ```
+      Expected: No changes displayed on TrackAScholar GUI and an error message will be prompted,
+      indicating applicants with duplicate names are not allowed.
+
+   1. Test case:
+      ```
+      add n/Alex Yeoh
+      m/Computer Science
+      ```
+      Expected: No changes displayed on TrackAScholar GUI and an error message will be prompted,
+      indicating an applicant must have specified phone number, email, scholarship name, application status,
+      as represented by the prefixes `p/`, `e/`, `s/` and `as` respectively.
+
+--------------------------------------------------------------------------------------------------------------------
+
+### Editing an applicant
+1. Editing an applicant in TrackAScholar while all applicants are displayed in GUI.
+    1. Prerequisites: List all applicants using the `list` command. One applicant in the list. The list contains
+       an applicant named `Alex Yeoh` with the following characteristics:
+       ```
+       Name: Alex Yeoh 
+       Major(s): Medicine
+       Application Status: pending
+       Scholarship Name: NUS Global Merit Scholarship
+       Email: alexyeoh@yahoo.com
+       Phone Number: 87438807
+       ```
+
+    1. Test case: `edit 1 n/Samuel Low e/samuelLow@gmail.com` <br>
+       Expected: The name and email of the first applicant in the list will be modified to `Samuel Low` and `samuelLow@gmail.com`
+       respectively, while the rest of the characteristics remains the same.
+
+    1. Test case: `edit 1 m/` <br>
+       Expected: The major(s) of the first applicant in the list will be 
+       removed completely, while the rest of the characteristics remains the same.
+
+    1. Test case: `edit 1 m/Mathemetics m/Engineering` <br>
+       Expected: The major(s) of the first applicant in the list will be
+       replaced by `Mathematics` and `Engineering` completely , while the rest of the characteristics remains the same.
+   
+    1. Test case: `edit 0 n/Dexter` <br>
+     Expected: No changes displayed on TrackAScholar GUI and an error message will be prompted,
+     indicating that `0` is an invalid index
+
+    1. Test case: `edit 50 n/Eric` <br>
+      Expected: No changes displayed on TrackAScholar GUI and an error message will be prompted,
+      indicating that `50` is out of bounds since the list contains only one applicant.
+
+    1. Other incorrect edit commands to try: `edit p/88125671` and `edit` <br>
```

Comment:
I think could have another manual test case where the user tries to edit an applicant's name to an already existing applicant's name.